vendor/plist_to_html/plist_to_html/PlistToHtml.py

```diff
@@ -59,12 +74,75 @@ def create(self, output_path, report_data):
         """
         Create html file with the given report data to the output path.
         """
+        self.generated_html_reports[output_path] = report_data['reports']
         content = self._layout.replace('<$REPORT_DATA$>',
                                        json.dumps(report_data))
 
         with open(output_path, 'w+') as html_output:
             html_output.write(content)
 
+    def create_index_html(self, output_dir):
+        """
+        Creates an index.html file which lists all available bugs which was
+        found in the processed plist files. This also creates a link for each
+        bug to the created html file where the bug can be found.
+        """
+
+        # Create table header.
+        table_reports = '''
+            <tr>
+              <th>File</th>
+              <th>Severity</th>
+              <th>Checker name</th>
+              <th>Message</th>
+              <th>Bug path length</th>
+            </tr>'''
+
+        # Sort reports based on severity levels.
+        report_data = []
+        for html_file in self.generated_html_reports:
+            for report in self.generated_html_reports[html_file]:
+                checker = report['checkerName']
+                report['severity'] =\
+                    self._severity_map.get(checker, 'UNSPECIFIED')
+                report_data.append({'html_file': html_file, 'report': report})
+        report_data = sorted(report_data,
+                             key=lambda d: d['report']['severity'])
+
+        # Create table lines.
+        for data in report_data:
+            html_file = data['html_file']
+            report = data['report']
+
+            events = report['events']
+            checker = report['checkerName']
+            severity = report['severity']
+
+            table_reports += '''
+              <tr>
+                <td>
+                  <a href="{0}#reportHash={1}">{2} @ Line {3}</a>
+                </td>
+                <td class="severity">
+                  <i class="severity-{4}"></i>
+                </td>
+                <td>{5}</td>
+                <td>{6}</td>
+                <td class="bug-path-length">{7}</td>
+              </tr>'''.format(os.path.basename(html_file),
+                              report['reportHash'],
+                              report['path'],
+                              events[-1]['line'],
+                              severity.lower() if severity else '',
```

Comment:
Why do we need this check? Shouldn't we always have a severity? (Unspecified if not provided)